### ISSUE TITLE
Another brew cask style error happened

### ISSUE THREAD

**Issue by RJHsiao** [Homebrew/homebrew-cask#48997]
#### General troubleshooting steps

- [ ] I have retried my command with `--force` and the issue is still present.
- [x] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
  - [x] None of the templates was appropriate for my issue, or I’m not sure.
- [x] I ran `brew update-reset && brew update` and retried my command.
- [x] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [x] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).

#### Description of issue

Get error when running command `brew cask style`

#### Command that failed

```
brew cask style
```

#### Output of command with `--verbose --debug`

```
$ sudo gem install rubocop-cask
Password:
Fetching: rubocop-cask-0.20.0.gem (100%)
Successfully installed rubocop-cask-0.20.0
Parsing documentation for rubocop-cask-0.20.0
Installing ri documentation for rubocop-cask-0.20.0
Done installing documentation for rubocop-cask after 0 seconds
1 gem installed

$ brew cask style --verbose --debug
==> Installing or updating 'rubocop-cask' gem
Fetching: public_suffix-3.0.2.gem (100%)
Error: installing into parent path bin/console of /usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.3.0/gems/public_suffix-3.0.2 is not allowed
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package.rb:439:in `block in mkdir_p_safe'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package.rb:433:in `each'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package.rb:433:in `reduce'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package.rb:433:in `mkdir_p_safe'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package.rb:379:in `block (2 levels) in extract_tar_gz'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package/tar_reader.rb:65:in `each'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package.rb:363:in `block in extract_tar_gz'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package.rb:472:in `block in open_tar_gz'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package.rb:469:in `wrap'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package.rb:469:in `open_tar_gz'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package.rb:362:in `extract_tar_gz'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package.rb:343:in `block (2 levels) in extract_files'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package/tar_reader.rb:65:in `each'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package.rb:340:in `block in extract_files'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package/file_source.rb:30:in `open'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package/file_source.rb:30:in `with_read_io'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/package.rb:337:in `extract_files'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/installer.rb:773:in `extract_files'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/installer.rb:295:in `install'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/resolver/specification.rb:97:in `install'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/request_set.rb:166:in `block in install'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/request_set.rb:156:in `each'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/request_set.rb:156:in `install'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/commands/install_command.rb:250:in `install_gem'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/commands/install_command.rb:300:in `block in install_gems'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/commands/install_command.rb:296:in `each'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/commands/install_command.rb:296:in `install_gems'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/commands/install_command.rb:203:in `execute'
/usr/local/Homebrew/Library/Homebrew/utils.rb:229:in `install_gem!'
/usr/local/Homebrew/Library/Homebrew/utils.rb:237:in `install_gem_setup_path!'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/style.rb:21:in `block in install_rubocop'
/usr/local/Homebrew/Library/Homebrew/utils.rb:405:in `capture_stderr'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/style.rb:19:in `install_rubocop'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/style.rb:11:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/abstract_command.rb:33:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:95:in `run_command'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:164:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:129:in `run'
/usr/local/Homebrew/Library/Homebrew/cmd/cask.rb:7:in `cask'
/usr/local/Homebrew/Library/Homebrew/brew.rb:100:in `<main>'
Error: Kernel.exit
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:175:in `exit'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:175:in `rescue in run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:152:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:129:in `run'
/usr/local/Homebrew/Library/Homebrew/cmd/cask.rb:7:in `cask'
/usr/local/Homebrew/Library/Homebrew/brew.rb:100:in `<main>'
```

#### Output of `brew cask doctor`

```
$ brew cask doctor
==> Homebrew-Cask Version
Homebrew-Cask 1.6.9-27-g7db0f97
Homebrew/homebrew-cask (git revision 3e7cf; last commit 2018-06-25)
==> macOS
10.13.6
==> SIP
Enabled
==> Java
10.0.1, 1.8.0_172
==> Homebrew-Cask Install Location
<NONE>
==> Homebrew-Cask Staging Location
/usr/local/Caskroom
==> Homebrew-Cask Cached Downloads
~/Library/Caches/Homebrew/Cask (4 files, 212.8MB)
==> Homebrew-Cask Taps:
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask (3981 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-versions (180 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-eid (11 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-fonts (1163 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-drivers (150 casks)
==> Contents of $LOAD_PATH
/usr/local/Homebrew/Library/Homebrew
/usr/local/Homebrew/Library/Homebrew/cask/lib
/Library/Ruby/Gems/2.3.0/gems/did_you_mean-1.0.0/lib
/Library/Ruby/Site/2.3.0
/Library/Ruby/Site/2.3.0/x86_64-darwin17
/Library/Ruby/Site/2.3.0/universal-darwin17
/Library/Ruby/Site
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0/x86_64-darwin17
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0/universal-darwin17
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/x86_64-darwin17
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/universal-darwin17
==> Environment Variables
LC_ALL="en_US.UTF-8"
PATH="/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/Homebrew/Library/Homebrew/shims/scm"
SHELL="/usr/local/bin/zsh"
```

#### Output of `brew tap`

```
homebrew/cask
homebrew/cask-drivers
homebrew/cask-eid
homebrew/cask-fonts
homebrew/cask-versions
homebrew/completions
homebrew/core
homebrew/dupes
homebrew/fuse
homebrew/php
homebrew/services
homebrew/x11
mpv-player/mpv
vitorgalvao/tiny-scripts
```

---

#### Ruby & Gem

Ruby:

```
$ which ruby
/usr/local/bin/ruby

$ which -a ruby
/usr/local/bin/ruby
/usr/bin/ruby

$ ruby --version
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin17]

$ /usr/local/bin/ruby --version
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin17]

$ /usr/bin/ruby --version
ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin17]
```

Gem: 

```
$ which gem
/usr/local/bin/gem

 $ which -a gem
/usr/local/bin/gem
/usr/bin/gem

$ gem --version
2.7.6

$ /usr/local/bin/gem --version
2.7.6

$ /usr/bin/gem --version
2.5.2.3
```

#### Output of `brew doctor`

```
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Ruby version 2.3.7 is unsupported on 10.13. Homebrew
is developed and tested on Ruby 2.3.3, and may not work correctly
on other Rubies. Patches are accepted as long as they don't cause breakage
on supported Rubies.

$ echo $?
1
```

P.S. I've got this error on my MacBook Pro Retina mid.2015 with macOS 10.13.6 **"Developer Beta 4"**, and it doesn't happened on my another MacBook Pro Retina with macOS 10.13.5 (current release version).

**Comment by mrleeio** [comment]
Getting the same error on a machine recently upgraded to 10.13.6.

Not sure if it causes any problems with my workflow yet.

**Comment by reitermarkus** [comment]
This was fixed by upgrading our portable Ruby to 2.3.7.
